spec/services/git_based_domain_import_service_spec.rb

```diff
@@ -102,4 +108,33 @@
       end
     end
   end
+
+  describe "#queue_refresh_and_import" do
+    include_context "import setup"
+    before do
+      allow(User).to receive(:current_user).and_return(user)
+    end
+
+    context "when git branches that match the given name do not exist" do
+      let(:git_branches) { [] }
+      let(:ref_name) { "the_tag_name" }
+      let(:ref_type) { "tag" }
+      let(:method_name) { 'import_git_url' }
+      let(:action) { 'Refresh and import git repository' }
+      let(:import_options) do
+        {
+          "git_url"   => git_repo.url,
+          "ref"       => ref_name,
+          "ref_type"  => ref_type,
+          "tenant_id" => 321
+        }
+      end
+
+      it "calls 'queue_import' with the correct options" do
+        expect(MiqTask).to receive(:generic_action_with_callback).with(task_options, queue_options).and_return(task.id)
+
+        expect(subject.queue_refresh_and_import(git_repo.url, ref_name, ref_type, 321)).to eq(task.id)
```

Comment:
I generally don't like having two `expect`s in a single `it` block, but I understand if we split them up, then there would need to be an `allow(MiqTask).to receive` line in a before block to handle the second expect. I think since there's nothing else really going on here that it's ok to leave it as is, but if there were more, then I'd lean towards splitting it.